### PR TITLE
refactor copyFiles function to remove fs-extra

### PR DESCRIPTION
**Pull request by lukasoppermann**
## Summary

Refactor `copyFilesAndFolders` to `copyFromDir`.

- It now only works with on source and one desitionation
- It can only copy the content of an entire folder

**Reason:** This is the only use case we have.

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change
- [ ] Verify the design tokens changed in this PR are expected using the diffing results in this PR

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -88,6 +88,7 @@ module.exports = {
       extends: ['plugin:@typescript-eslint/recommended', 'prettier'],
       rules: {
         'no-shadow': 'off',
+        'import/no-nodejs-modules': 'off',
```
Comment (rezrah) [inline .eslintrc.js:91]: Guessing you're wanting to turn this off for server-side scripts? If so, I'd recommend creating a more targetted override set. You've added this one into `['**/*.{ts,tsx}']`, which includes files that can be run client-side, where node modules shouldn't be run.

Comment (lukasoppermann) [reply .eslintrc.js:91]: Okay, sounds good. How do I do this?

Should I just do this for `['config/**/*.ts','script/**/*.ts']`?

File: config/utilities/copyFromDir.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * @param destination path
+ * @returns promise
+ */
+export const copyFromDir = async (source: string, destination: string) => {
+  // adjust trailing slash
+  const dest = `${destination.replace(/\/$/, '')}/`
+  const src = `${source.replace(/\/$/, '')}`
+  execSync(`mkdir -p ${dest}`)
+  return exec(`cp -R ${src}/* ${dest}`)
+}
```
Comment (rezrah) [inline config/utilities/copyFromDir.ts:16]: just curious if you'd considered these alternatives...

a) using `fs` as it has better process parallelisation 
b) running natively through a shell script, to avoiding blocking the event loop and saturating the JS thread with something like `execSync`

File: .eslintrc.js (hunk 1 above)
Comment (rezrah) [reply .eslintrc.js:91]: yeah that sounds good for now, and can be adjusted later if needed 👍

File: config/utilities/copyFromDir.ts (hunk 2 above)
Comment (lukasoppermann) [reply config/utilities/copyFromDir.ts:16]: No, neither of those.

I would like to keep it in the script, which is why I think **b)** is not so good. 

I am happy to do **a)** if you can point me to where I can learn how to. This is just the way I know. 🙃

Comment (rezrah) [reply config/utilities/copyFromDir.ts:16]: to expand on a) I think you can pass `recursive` option to `fs.cp`, which can achieve the same thing. I'm totally onboard with dropping fs-extra for something like this though.

Comment (lukasoppermann) [reply config/utilities/copyFromDir.ts:16]: Wait, you mean:
- fs.readdirSync
- loop → fs.copyFile?

Comment (lukasoppermann) [reply config/utilities/copyFromDir.ts:16]: `fs.cp` is experimental though, is this not a problem?

Comment (rezrah) [reply config/utilities/copyFromDir.ts:16]: either of the approaches you mentioned would be fine. We're at node 18 now, so not long to go to 19 where fs.cp comes in. Happy to play it safe though and use fs.copyFile instead.

Comment (lukasoppermann) [reply config/utilities/copyFromDir.ts:16]: If we go with `fs.cp` do we need to add node@18 to our package.json? We currently don't have node in there. But we should maybe add it at least as a peer-dep? 🤔

Comment (rezrah) [reply config/utilities/copyFromDir.ts:21]: Yeah it'll run the background, but the problem there may be that you'll run into race conditions if your function returns too early. If it works, best to leave as-is. I think it's fine. Alternatively, add the await or use `copyFileSync` maybe?

**Review by rezrah — APPROVED** [review]
Thanks for following up on removing fs-extra 🙇 ... latest updates look great 🤩

File: config/utilities/copyFromDir.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/* istanbul ignore file */
+import {copyFile, readdir, mkdir} from 'fs/promises'
+/**
+ * copyFromDir
+ * @description Copies all files from source folrder to destination
```
Comment (rezrah) [inline config/utilities/copyFromDir.ts:5]:
```suggestion
 * @description Copies all files from source folder to destination
```

**Pull request merged**